local_modules/execer/execer.cc

```diff
@@ -15,17 +15,56 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <time.h>
 #include <node.h>
+#include <sstream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
 #include <sys/socket.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <v8.h>
 #include <vector>
 
 using namespace v8;
 
+bool write_response(int fd) {
+	// This is what Nginx does.
+	char buf[2048];
+	time_t now = time(0);
+	struct tm gm = *gmtime(&now);
+	if (strftime(buf, sizeof buf, "%a, %d %b %Y %H:%M:%S %Z", &gm) <= 0) {
+		fprintf(stderr, "strftime's output didn't fit in 1024 bytes\n");
+		exit(1);
+	}
+	std::ostringstream response;
+	response << "HTTP/1.0 200 OK\n" <<
+	"Date: " << buf << "\n" <<
+	"Content-Length: 23\n" <<
+	"Content-Type: text/plain; charset=utf-8\n\n" <<
+	"User function is ready\n";
+	std::string res = response.str();
+
+	while (true) {
+		ssize_t result = write(fd, res.c_str(), res.length());
+		if (result == 0) {
+			// Socket is writable, but EOF.
+			return true;
+		}
+		if (result != -1) {
+			fprintf(stderr, "wrote %zd of %lu byte(s) to FD %d, ", result, res.length(), fd);
+			res = res.substr(static_cast<unsigned long>(result));
+			fprintf(stderr, "%lu byte(s) left\n", res.length());
```

Comment:
As of now, I find them useful and they get eaten on GCF. I will re-visit when I figure out logging.